src/Features/Core/Portable/ExternalAccess/UnitTesting/API/UnitTestingIncrementalAnalyzerProvider.cs

```diff
@@ -35,5 +37,24 @@ public void Reanalyze()
                 solutionCrawlerService.Reanalyze(_workspace, analyzer, projectIds: null, documentIds: null, highPriority: false);
             }
         }
+
+        public static UnitTestingIncrementalAnalyzerProvider? TryRegister(Workspace workspace, string analyzerName, IUnitTestingIncrementalAnalyzerProviderImplementation provider)
+        {
+            var solutionCrawlerRegistrationService = workspace.Services.GetService<ISolutionCrawlerRegistrationService>();
+            if (solutionCrawlerRegistrationService == null)
+            {
+                return null;
+            }
+
+            var analyzerProvider = new UnitTestingIncrementalAnalyzerProvider(workspace, provider);
+
+            var metadata = new IncrementalAnalyzerProviderMetadata(
+                analyzerName,
+                highPriorityForActiveFile: false,
+                new[] { workspace.Kind });
```

Comment:
@sharwell @tmat The main thing that changed in this PR is the above line. Previously this was hardcoded to `WorkspaceKind.RemoteWorkspace`.

I am assuming the `workspace.Kind` will always be correctly set on workspaces (i.e. `WorkspaceKind.RemoteWorkspace` in case of OOP workspace and `WorkspaceKind.HostWorksoace` in case of local ones - code on the UT side will call this function with `VisualStudioWorkspace` in the local case and I'm assuming the workspace returned in the OOP case by the call to `GetWokspace()` below will always have `WorkspaceKind.RemoteWorkspace`).

❓ Is this a valid assumption?

(Unfortunately I am running into problems building Roslyn repo atop latest VS main so I haven't been able to test this change. This is the main reason I was holding off on merging this - but now that its merged, just wanted to make sure it won't break anything 😄 )

Reply:
Yes, this should be a valid assumption